src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int CAPACITY_DIVISOR = 2;
+    private int size;
+    private Object[] data;
+
+    public ArrayList() {
+        this.size = 0;
+        this.data = new Object[DEFAULT_CAPACITY];
```

Comment:
1. why do you use special word `this`? no need to do it, better read when we need to use it
2. data array should be parameterized
```suggestion
        data = (T[]) new Object[DEFAULT_CAPACITY];
```